scaladoc/src/dotty/tools/scaladoc/site/templates.scala

```diff
@@ -110,11 +121,51 @@ case class TemplateFile(
       case other => other
 
     // Library requires mutable maps..
-    val mutableProperties = new JHashMap(ctx.properties.transform((_, v) => asJavaElement(v)).asJava)
+    val mutableProperties = new JHashMap[String, Any](
+      ctx.properties.transform((_, v) => asJavaElement(v)).asJava
+    )
+
+
+    val dataPath =  ssctx.root.toPath.resolve("_data")
+
+    // Load the data from yaml file in _data folder
+    val dataMap = DataLoader().loadDataDirectory(dataPath.toString)
+
+
+    val siteData = new JHashMap[String, Any]()
+    siteData.put("data",dataMap)
+
+
+    // assign the the path for Include Tag
+    val includePath =  ssctx.root.toPath.resolve("_includes")
+    IncludeTag.setDocsFolder(includePath.toString)
+
+    val configLoader = new SiteConfigLoader()
+    val configMap = configLoader.loadConfig(ssctx.root.toPath.toString)
+    val siteConfig: java.util.Map[String, Any] = configMap.convertToJava
+
+
+
+    LanguagePickerTag.setConfigValue(configMap)
+
+
+
+    siteData.put("config",siteConfig)
+
+    mutableProperties.put("site",siteData)
+
+//    val parseSettings = ParseSettings.Builder().withFlavor(Flavor.JEKYLL).build().withBlock(AltDetails())
+    val liqpParser = TemplateParser.Builder()
```

Comment:
So at some point the connection should be made between enabling the experimental settings flag and enabling these new features